src/hooks/useMixpanel.ts

```diff
@@ -0,0 +1,22 @@
+// eslint-disable-next-line no-restricted-imports
+import mixpanel, { Dict, RequestOptions, Callback } from "mixpanel-browser";
+import { useMemo } from "react";
+import { MIXPANEL_PROJECT_TOKEN } from "secrets";
+
+const noopTrack: (
+  event_name: string,
+  properties?: Dict,
+  optionsOrCallback?: RequestOptions | Callback,
+  callback?: Callback
+) => void = () => {};
+
+export const useMixpanel = () => {
+  const mixpanelMemo = useMemo(
+    () => ({
+      track: MIXPANEL_PROJECT_TOKEN ? mixpanel.track.bind(mixpanel) : noopTrack,
```

Comment:
Mixpanel doesn't seem to provide an easy way to know if it's initialized, but I don't want the conditional spread through the code / error prone / repeated. So added a hook to access it, added linting etc.

https://stackoverflow.com/questions/32203420/check-if-mixpanel-library-has-been-loaded